### PR TITLE
Feature/create orders

### PR DESCRIPTION
**Pull request by Yuliana1907**

### CODE REVIEW

File: src/components/_templates/AddOrders/rules.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
+export const orderFormRules = () => {
```
Comment (Georgi-Filipov) [inline src/components/_templates/AddOrders/rules.ts:1]: Add the inly one rule "This field is required!"

File: src/components/_templates/constants.tsx [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    latitude: string
+    longitude: string
+    address: string
+    description: string
+  }
+}
+
+export const initialOrderData = (orderData: OrderData): IOrder => ({
+  id: (Math.random() + 1).toString(36).substring(2),
+  user_id: (Math.random() + 1).toString(36).substring(2),
```
Comment (Georgi-Filipov) [inline src/components/_templates/constants.tsx:28]: user_id - is a current user id

File: src/components/_templates/constants.tsx [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  price_min: orderData.price_min,
+  price_max: orderData.price_max,
+  location: {
+    latitude: orderData.location.latitude,
+    longitude: orderData.location.longitude,
+  },
+  address: orderData.address,
+  description: orderData.description,
+  created_at: getDateNow(),
+  path: [
```
Comment (Georgi-Filipov) [inline src/components/_templates/constants.tsx:40]: Check, "id", "status", "created_at", "path" should be added automatically from BE side

File: src/components/_templates/AddOrders/AddOrders.tsx [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+              </Form.Item>
+            </Input.Group>
+          </Form.Item>
+          <Form.Item label="Address" name="address" rules={[addressRules]}>
+            <Input
+              autoComplete="new-password"
+              placeholder="Please input address"
+            />
+          </Form.Item>
+          <Button>{CONSTANTS_TEXT.USE_CURRENT}</Button>
```
Comment (Georgi-Filipov) [inline src/components/_templates/AddOrders/AddOrders.tsx:74]: when user click "USE CURRENT" - current address and location should be added to form fields

File: src/components/_atoms/Modal/Modal.tsx [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+  title: string
+  isOpen: boolean
+  onClose: () => void
+  children: ReactNode
+}
+
+export const Modal = ({ title, isOpen, children }: ModalProps) => {
+  return isOpen ? (
+    <div className="modal">
+      <div className="modal_wrapper" />
```
Comment (AlehSabadashGmail) [inline src/components/_atoms/Modal/Modal.tsx:14]: why are you using custom modal? What about modal from antd and it is not a atom, modal consist of header, buttos, it can be molecules or organisms

File: src/components/_pages/HomePage/HomePage.tsx [hunk 6]
```diff
@@ -1,5 +1,11 @@
 import React from 'react'
+import { AddOrders } from '../../_templates'
 
 export const HomePage = () => {
-  return <div className="home-page">Home page</div>
+  return (
+    <>
+      <div className="home-page">Home page</div>
+      <AddOrders />
```
Comment (AlehSabadashGmail) [inline src/components/_pages/HomePage/HomePage.tsx:8]: why did you add it here, I thought that AddOrders feature should be new route

File: src/components/_templates/AddOrders/AddOrders.tsx [hunk 7]
```diff
@@ -0,0 +1,110 @@
+import { Button, Form, Input, InputNumber } from 'antd'
+import TextArea from 'antd/lib/input/TextArea'
+import React, { useState } from 'react'
+import api from '../../../helper/api'
+import { IOrder } from '../../../redux/orders/type'
+import { Modal } from '../../_atoms/Modal'
+import { currentUser, initialOrderData, OrderData } from '../constants'
+import './AddOrders.scss'
```
Comment (AlehSabadashGmail) [inline src/components/_templates/AddOrders/AddOrders.tsx:8]: if you do not use styles you do not need to create file for that

File: src/components/_templates/AddOrders/rules.ts [hunk 8]
```diff
@@ -0,0 +1,8 @@
...
+  return {
+    rule: {
```
Comment (AlehSabadashGmail) [inline src/components/_templates/AddOrders/rules.ts:3]: Does it make sense to create common folder, add file with rules and put all common rules in this file ?

File: src/components/_templates/constants.tsx [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+    longitude: string
+  }
+}
+
+export const initialOrderData = (orderData: OrderData): IOrder => ({
+  // send the id of the logged user
+  user_id: currentUser.id,
+  product_name: orderData.product_name,
+  price_min: orderData.price_min,
+  price_max: orderData.price_max,
```
Comment (AlehSabadashGmail) [inline src/components/_templates/constants.tsx:21]: can we use here spread operator , so much code

File: src/components/_templates/AddOrders/AddOrders.tsx [hunk 10]
```diff
@@ -0,0 +1,111 @@
...
+  }
+
+  const createOrders = () => {
+    api()
+      .post('orders', ordersValues)
+      .then((response) => console.log(response.data))
+  }
+
+  const onFinish = (values: OrderData) => {
+    setOrdersValues(initialOrderData(values))
```
Comment (AlehSabadashGmail) [inline src/components/_templates/AddOrders/AddOrders.tsx:32]: useless state for ordersValue

File: src/components/_templates/constants.tsx [hunk 11]
```diff
@@ -0,0 +1,38 @@
+import { IOrder } from '../../redux/orders/type'
+
+export type OrderData = {
+  product_name: string
+  price_min: number
+  price_max: number
+  address: string
+  description: string
```
Comment (AlehSabadashGmail) [inline src/components/_templates/constants.tsx:8]: why did you put Type in constants

File: src/components/_templates/AddOrders/AddOrders.tsx [hunk 12]
```diff
@@ -0,0 +1,104 @@
...
+  const handleCancel = () => {
+    setVisible(false)
+    form.resetFields()
+  }
+
+  const onFinish = (values: OrderData) => {
+    if (values) {
+      api()
+        .post('orders', { ...values, user_id: CURRENT_USER.id })
+        .then((response) => console.log(response.data))
```
Comment (Georgi-Filipov) [inline src/components/_templates/AddOrders/AddOrders.tsx:27]: console.log??

File: src/components/_templates/AddOrders/AddOrders.tsx [hunk 13]
```diff
@@ -0,0 +1,104 @@
...
+    setVisible(true)
+  }
+
+  const handleCancel = () => {
+    setVisible(false)
+    form.resetFields()
+  }
+
+  const onFinish = (values: OrderData) => {
+    if (values) {
```
Comment (Georgi-Filipov) [inline src/components/_templates/AddOrders/AddOrders.tsx:24]: "values" can not be empty if validation is correct
